panic: not struct

### Nuclei version:
dev

### Current Behavior:
```console
go install -v github.com/projectdiscovery/nuclei/v2/cmd/nuclei@dev
sudo go/bin/nuclei -no-interactsh -rl 50 -u 172.30.30.30:5666

                      _
   __    _/ /  (_)
  /  / / / / / /  / /
 / / / / /_/ / // /  / /
// //_,/_//___//   v2.9.3-dev

                projectdiscovery.io

[INF] Current nuclei version: v2.9.3-dev (development)
[INF] Current nuclei-templates version: v9.4.3 (latest)
[INF] New templates added in latest release: 55
[INF] Templates loaded for current scan: 5901
[INF] Targets loaded for current scan: 1
[INF] Running httpx on input host
[INF] Found 0 URL from httpx
[INF] Templates clustered: 1053 (Reduced 974 Requests)
panic: not struct

goroutine 1383 [running]:
github.com/fatih/structs.strctVal({0x1b634e0?, 0x0?})
        /home/myhomedir/go/pkg/mod/github.com/fatih/structs@v1.1.0/structs.go:437 +0xdf
github.com/fatih/structs.New(...)
        /home/myhomedir/go/pkg/mod/github.com/fatih/structs@v1.1.0/structs.go:30
github.com/projectdiscovery/nuclei/v2/pkg/protocols/ssl.(*Request).ExecuteWithResults(0xc000042fc0, 0xc0029819c0, 0xc007319800?, 0xc007319830?, 0xc007319860)
/home/myhomedir/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.9.1-0.20230502181956-449afc0c5c30/pkg/protocols/ssl/ssl.go:267 +0x1405
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/executer.(Executer).Execute(0xc004164800, 0xc002981580)
        /home/myhomedir/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.9.1-0.20230502181956-449afc0c5c30/pkg/protocols/common/executer/executer.go:81 +0x383
github.com/projectdiscovery/nuclei/v2/pkg/core.(Engine).executeTemplateWithTargets.func2.1(0x1605e8a?, 0x0?, 0xc002981460)
        /home/myhomedir/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.9.1-0.20230502181956-449afc0c5c30/pkg/core/executors.go:128 +0x296
created by github.com/projectdiscovery/nuclei/v2/pkg/core.(*Engine).executeTemplateWithTargets.func2
        /home/myhomedir/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.9.1-0.20230502181956-449afc0c5c30/pkg/core/executors.go:105 +0x4f8
```

### Expected Behavior:
no crash



### Anything else:
<!-- Links? References? Screenshots? Anything that will give us more context about the issue that you are encountering! -->

--- a/v2/pkg/protocols/ssl/ssl.go
+++ b/v2/pkg/protocols/ssl/ssl.go
@@ -255,17 +255,36 @@ func (request *Request) ExecuteWithResults(input *contextargs.Context, dynamicVa
 	data["template-id"] = requestOptions.TemplateID
 	data["template-info"] = requestOptions.TemplateInfo
 
+	// if response is not struct compatible, error out
+	if !structs.IsStruct(response) {
+		return errorutil.NewWithTag("ssl", "response cannot be parsed into a struct: %v", response)
+	}
+
 	// Convert response to key value pairs and first cert chain item as well
 	responseParsed := structs.New(response)
 	for _, f := range responseParsed.Fields() {
+		if !f.IsExported() {
+			// if field is not exported f.IsZero() , f.Value() will panic
+			continue
+		}
 		tag := utils.CleanStructFieldJSONTag(f.Tag("json"))
 		if tag == "" || f.IsZero() {
 			continue
 		}
 		data[tag] = f.Value()
 	}
+
+	// if certificate response is not struct compatible, error out
+	if !structs.IsStruct(response.CertificateResponse) {
+		return errorutil.NewWithTag("ssl", "certificate response cannot be parsed into a struct: %v", response.CertificateResponse)
+	}
+
 	responseParsed = structs.New(response.CertificateResponse)
 	for _, f := range responseParsed.Fields() {
+		if !f.IsExported() {
+			// if field is not exported f.IsZero() , f.Value() will panic
+			continue
+		}
 		tag := utils.CleanStructFieldJSONTag(f.Tag("json"))
 		if tag == "" || f.IsZero() {
 			continue

